internal/receiver/prometheusremotewritereceiver/internal/prometheus_spec_utils.go

```diff
@@ -0,0 +1,87 @@
+// Copyright Splunk, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/prometheus/prometheus/prompb"
+)
+
+// GetBaseMetricFamilyName uses heuristics to determine the metric family of a given metric, by
+// removing known suffixes for Sum/Counter, Histogram and Summary metric types.
+// While not strictly enforced in the protobuf, prometheus does not support "colliding"
+// "metric family names" in the same write request, so this should be safe
+// https://prometheus.io/docs/practices/naming/
+// https://prometheus.io/docs/concepts/metric_types/
+func GetBaseMetricFamilyName(metricName string) string {
+	suffixes := []string{"_count", "_sum", "_bucket", "_created", "_total"}
+	for _, suffix := range suffixes {
+		if strings.HasSuffix(metricName, suffix) {
+			return strings.TrimSuffix(metricName, suffix)
+		}
+	}
+
+	return metricName
+}
+
+// ExtractMetricNameLabel Finds label corresponding to metric name
+func ExtractMetricNameLabel(labels []prompb.Label) (string, error) {
+
+	for _, label := range labels {
+		if label.Name == "__name__" {
+			return label.Value, nil
+		}
+	}
```

Comment:
This convention is used throughout so I'd think a `getLabel(labels []prompb.Label) (string, bool)` helper could be useful